src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Controllers/Api/ReleasesControllerTests.cs

```diff
@@ -296,24 +299,89 @@ public async Task GetDeleteDataFilePlan()
 
             releaseService
-                .Setup(s => s.GetDeleteReleasePlan(_releaseVersionId))
+                .Setup(s => s.GetDeleteReleaseVersionPlan(_releaseVersionId, It.IsAny<CancellationToken>()))
                 .ReturnsAsync(deleteReleasePlan);
 
             var controller = BuildController(releaseService: releaseService.Object);
 
-            var result = await controller.GetDeleteReleasePlan(_releaseVersionId);
+            var result = await controller.GetDeleteReleaseVersionPlan(_releaseVersionId, It.IsAny<CancellationToken>());
             VerifyAllMocks(releaseService);
 
             result.AssertOkResult(deleteReleasePlan);
         }
 
+        [Fact]
+        public async Task DeleteReleaseVersion_Returns_NoContent()
+        {
+            var releaseService = new Mock<IReleaseService>(Strict);
+
+            var fileId = Guid.NewGuid();
+
+            releaseService
+                .Setup(service => service.DeleteReleaseVersion(_releaseVersionId))
+                .ReturnsAsync(Unit.Instance);
+
+            var controller = BuildController(releaseService: releaseService.Object);
+
+            var result = await controller.DeleteReleaseVersion(_releaseVersionId);
+            VerifyAllMocks(releaseService);
+
+            Assert.IsAssignableFrom<NoContentResult>(result);
+        }
+
+        [Fact]
+        public async Task DeleteReleaseVersion_Returns_NotFound()
+        {
+            var releaseService = new Mock<IReleaseService>(Strict);
+
+            var fileId = Guid.NewGuid();
+
+            releaseService
+                .Setup(service => service.DeleteReleaseVersion(_releaseVersionId))
+                .ReturnsAsync(new NotFoundResult());
+
+            var controller = BuildController(releaseService: releaseService.Object);
+
+            var result = await controller.DeleteReleaseVersion(_releaseVersionId);
+            VerifyAllMocks(releaseService);
+
+            result.AssertNotFoundResult();
+        }
+
+        [Fact]
+        public async Task DeleteReleaseVersion_Returns_ValidationProblem()
+        {
+            var releaseService = new Mock<IReleaseService>(Strict);
+
+            var fileId = Guid.NewGuid();
+
+            releaseService
+                .Setup(service => service.DeleteReleaseVersion(_releaseVersionId))
+                .ReturnsAsync(ValidationUtils.ValidationResult(new ErrorViewModel
+                {
+                    Code = "error code",
+                    Path = "error path"
+                }));
+
+            var controller = BuildController(releaseService: releaseService.Object);
+
+            var result = await controller.DeleteReleaseVersion(_releaseVersionId);
+            VerifyAllMocks(releaseService);
+
+            var validationProblem = result.AssertBadRequestWithValidationProblem();
+
+            validationProblem.AssertHasError(
+                expectedPath: "error path",
+                expectedCode: "error code");
+        }
```

Comment:
These tests are the the old school non-integration controller tests which are essentially useless. Our controllers are usually super thin and the main logic resides in the service that's called. As we mock out the service, these tests basically prove nothing and have next to zero value.

We've previously agreed in dev team meetings that we wouldn't continue writing these styles of tests.

Of course, we can keep them now you've written them, but in the future would advise against spending any time working on these types of tests. Instead, either focus on the service tests (where the value actually is) or convert to integration tests. The choice should depend on what's most appropriate for that PR.